com.unity.shadergraph/Editor/Importers/ShaderSubGraphImporter.cs

```diff
@@ -177,6 +191,10 @@ static void ProcessSubGraph(SubGraphAsset asset, GraphData graph)
             List<AbstractMaterialNode> nodes = new List<AbstractMaterialNode>();
             NodeUtils.DepthFirstCollectNodesFromNode(nodes, outputNode);
 
+            // flag the used nodes so we can filter out errors from unused nodes
+            foreach (var node in nodes)
+                node.SetUsedByGenerator();
```

Comment:
similar to shadergraphs, this is the list of "actually used nodes", so we just push that down to the nodes themselves as a flag.